### PR TITLE
Fix a few FIXMEs

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: source/dealii/amge_host.cc [hunk 1]
```diff
@@ -27,3 +28,59 @@ template class mfmg::AMGe_host<
...
+mfmg::AMGe_host<3, mfmg::DealIIMatrixFreeMeshEvaluator<3>,
+                dealii::LinearAlgebra::distributed::Vector<double>>::
+    compute_local_eigenvectors(
+        unsigned int n_eigenvectors, double tolerance,
+        dealii::parallel::distributed::Triangulation<3> const
+            &agglomerate_triangulation,
+        std::map<typename dealii::Triangulation<3>::active_cell_iterator,
+                 typename dealii::DoFHandler<3>::active_cell_iterator> const
+            &patch_to_global_map,
+        mfmg::DealIIMatrixFreeMeshEvaluator<3> const &evaluator, int) const;
```
Comment (dalg24) [inline source/dealii/amge_host.cc]: Consider using a macro here
```C++
#define INSTANTIATE_COMPULTE_LOCAL_EIGENVECTORS(DIM, MESH_EVALUATOR) \
        template std::tuple<std::vector<std::complex<double>>, \
	                    std::vector<dealii::Vector<double>>, std::vector<double>, \
	                    std::vector<dealii::types::global_dof_index>> \
	mfmg::AMGe_host<DIM, MESH_EVALUATOR<DIM>, \
	                dealii::LinearAlgebra::distributed::Vector<double>>:: \
	    compute_local_eigenvectors( \
	        unsigned int n_eigenvectors, double tolerance, \
	        dealii::parallel::distributed::Triangulation<DIM> const \
	            &agglomerate_triangulation, \
	        std::map<typename dealii::Triangulation<DIM>::active_cell_iterator, \
	                 typename dealii::DoFHandler<DIM>::active_cell_iterator> const \
	            &patch_to_global_map, \
	        MESH_EVALUATOR<DIM> const &evaluator, int) const;

INSTANTIATE_COMPULTE_LOCAL_EIGENVECTORS(2, mfmg::DealIIMeshEvaluator)
INSTANTIATE_COMPULTE_LOCAL_EIGENVECTORS(3, mfmg::DealIIMeshEvaluator)
INSTANTIATE_COMPULTE_LOCAL_EIGENVECTORS(2, mfmg::DealIIMatrixFreeMeshEvaluator)
INSTANTIATE_COMPULTE_LOCAL_EIGENVECTORS(3, mfmg::DealIIMatrixFreeMeshEvaluator)
```

(Haven't actually tried, might need some adjustments)

File: source/dealii/amge_host.cc [hunk 2]
```diff
@@ -27,3 +28,24 @@ template class mfmg::AMGe_host<
 template class mfmg::AMGe_host<
     3, mfmg::DealIIMatrixFreeMeshEvaluator<3>,
     dealii::LinearAlgebra::distributed::Vector<double>>;
+
+#define INSTANTIATE_COMPUTE_LOCAL_EIGENVECTORS(DIM, MESH_EVALUATOR)            \
```
Comment (Rombur) [inline source/dealii/amge_host.cc]: We've put all the instantiations macro in `instantiation.hpp`

File: source/dealii/amge_host.cc [hunk 3]
```diff
@@ -27,3 +28,24 @@ template class mfmg::AMGe_host<
...
+          unsigned int n_eigenvectors, double tolerance,                       \
+          dealii::parallel::distributed::Triangulation<DIM> const              \
+              &agglomerate_triangulation,                                      \
+          std::map<                                                            \
+              typename dealii::Triangulation<DIM>::active_cell_iterator,       \
+              typename dealii::DoFHandler<DIM>::active_cell_iterator> const    \
+              &patch_to_global_map,                                            \
+          MESH_EVALUATOR<DIM> const &evaluator, int) const;
+
+INSTANTIATE_COMPUTE_LOCAL_EIGENVECTORS(2, mfmg::DealIIMeshEvaluator)
```
Comment (Rombur) [inline source/dealii/amge_host.cc]: You can use `BOOST_PP_REPEAT_FROM_TO` to reduce the number of instantiations

File: source/dealii/amge_host.cc (hunk 2 above)
Comment (masterleinad) [reply source/dealii/amge_host.cc]: Does it make sense to move this over? Opposed to all the other macros in `instantiation.hpp` this pretty specific.

**Pull request merged**